### ISSUE TITLE
Example code is susceptible to username hijacking

### ISSUE THREAD

**Issue by kgiszewski** [Shazwazza/UmbracoIdentity#87]
While I know the code you are providing in the views/controller is sample, the code is susceptible to allowing for a user to create an Umbraco account for an email/username which is not apart of the claims.

I realize it's up to the developer to look out for these sorts of situations but Identity is also a "not easy to learn" topic. I recently worked on a project where a malicious user could associate a different email by manipulating the hidden fields.

My suggestion is to discourage "picking your own username" and to discourage hidden fields for the username". Most developers are just happy it works, but I think this leaves a site open to misuse.

Scenario:
- User authenticates against 3rd-Party
- User is taken to the `ExternalLoginConfirmation` where the user can choose his/her username
- I can then choose `some_account@not_mine.com` or a hidden field is used for the user/email (which I can manipulate client side)
- Identity will then associate my login with an account that is not mine with potential malicious outcome

Suggestions:
- Change the sample code to not encourage the developer to ask for a username
- Don't rely on a hidden field for the email/username either
- Use information from the claims to create the username
- Change the following code inside `ExternalLoginConfirmation` to only take information from the claims and not the model:

![image](https://user-images.githubusercontent.com/4653907/39445417-f77c1f26-4c88-11e8-84a7-5b2966c279f3.png)

https://github.com/Shazwazza/UmbracoIdentity/blob/master/src/UmbracoIdentity.Web/App_Code/Controllers/UmbracoIdentityAccountController.cs#L138-L143

To sum up, thanks for writing this, just wanted to pass on my 2 cents.

I changed a project that I was working on to only use information provided via claims and disallowed using any sort of hidden fields on the form.

**Comment by kgiszewski** [comment]
Sure thing, I'll try to get you something 👍

**Comment by Shazwazza** [comment]
I've investigated this now and figured out what you are referring to. IIRC what happened was that a lot of this code is borrowed from the ASP.Net original identity 2.x snippets but since our members generally use an email address as their username, that snippet was updated whereas i think the original identity snippet actually uses a true username instead of an email.

In any case, i've simplified this whole thing, there will be no screen to confirm your username or email, it will just create your account with the email supplied from the claims and if any of that fails will redirect to an error page with the details. I'll get this released asap.

**Comment by kgiszewski** [comment]
Cool. I forgot that I was supposed to PR this. Glad I made some sense 😊

Thanks for the update.
